pkg/api/plugins.go

```diff
@@ -151,7 +151,10 @@ func (hs *HTTPServer) UpdatePluginSetting(c *models.ReqContext, cmd models.Updat
 	pluginID := c.Params(":pluginId")
 
 	if app := hs.PluginManager.GetApp(pluginID); app == nil {
-		return response.Error(404, "Plugin not installed", nil)
+		vcs := hs.PluginManager.VersionedControlStorage()
+		if vcs == nil || vcs.Id != pluginID {
+			return response.Error(404, "Plugin not installed", nil)
+		}
```

Comment:
Not sure about that... I think I'll create a new function in the manager to retrieve all plugins that are allow to have settings.